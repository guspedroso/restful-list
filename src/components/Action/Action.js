import React, { useState } from 'react';
import { Button, Container, Row, Col } from 'react-bootstrap';
import './Action.css';

const Action = (props) => {
    const [removeView, setRemoveView] = useState(false);

    const { hide = false, open = false, disabled = false,
            toggleOnly = false, actionType = 'Edit', outerClass = '',
            handleCreate, handleUpdate, handleRemove, handleToggle } = props;

    const handleRemoveView = () => setRemoveView(!removeView);

    return (
        <div className={outerClass}>
            { hide ? null :
            !open || toggleOnly ?
            <Button className="action-button pull-right" variant="primary" onClick={handleToggle} disabled={disabled}>
                {actionType}
            </Button> :
            removeView ?
            <Container>
                <Row>
                    <Col className="action-col" xs='6' sm='7'>
                        Are you sure you want to remove?
                    </Col>
                    <Col className="action-col" xs='6' sm='5'>
                        <Button className="action-button pull-right" variant="primary" onClick={handleRemove} disabled={disabled}>
                            Yes
                        </Button>
                        <Button className="action-button pull-right" variant="secondary" onClick={handleRemoveView} disabled={disabled}>
                            No
                        </Button>
                    </Col>
                </Row>
            </Container> :
            <Container>
                <Row>
                    <Col className="action-col">
                        { !!handleUpdate &&
                        <Button className="action-button pull-right" variant="primary" onClick={handleUpdate} disabled={disabled}>
                            Update
                        </Button> }
                        { !!handleCreate &&
                        <Button className="action-button pull-right" variant="primary" onClick={handleCreate} disabled={disabled}>
                            Create
                        </Button> }
                        <Button className="action-button pull-right" variant="secondary" onClick={handleToggle} disabled={disabled}>
                            Cancel
                        </Button>
                        { !!handleRemove &&
                        <Button className="action-button pull-right" variant="danger" onClick={handleRemoveView} disabled={disabled}>
                            Remove
                        </Button> }
                    </Col>
                </Row>
            </Container> }
        </div>
    );
}

export default Action;
