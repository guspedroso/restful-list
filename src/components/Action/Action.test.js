import React from 'react';
import ReactDOM from 'react-dom';
import Action from './Action';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Action />, div);
    ReactDOM.unmountComponentAtNode(div);
});

/*it('toggling the remove button works', () => {
    const wrapper = shallow(<Action />);
    const instance = wrapper.instance();

    expect(wrapper.state('removeView')).toBe(false);

    instance.toggleRemove();

    expect(wrapper.state('removeView')).toBe(true);
});*/

test('Snapshot works', () => {
    let component = renderer.create(
        <Action/>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Disabled snapshots work', () => {
    let component = renderer.create(
        <Action 
            disabled={true}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            disabled={false}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Open snapshots work', () => {
    let component = renderer.create(
        <Action 
            open={true}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={false}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Hide snapshots work', () => {
    let component = renderer.create(
        <Action 
            hide={true}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            hide={false}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Action types snapshots work', () => {
    let component = renderer.create(
        <Action 
            actionType={'Test'}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Outer class snapshots work', () => {
    let component = renderer.create(
        <Action 
            outerClass={'test'}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Open with different combinations of functions snapshots work', () => {
    let component = renderer.create(
        <Action 
            open={true}
            createAction={() => console.warn('create')}
        />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            updateAction={() => console.warn('update')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            removeAction={() => console.warn('remove')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            createAction={() => console.warn('create')}
            updateAction={() => console.warn('update')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            createAction={() => console.warn('create')}
            removeAction={() => console.warn('remove')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            updateAction={() => console.warn('update')}
            removeAction={() => console.warn('remove')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
        <Action 
            open={true}
            createAction={() => console.warn('create')}
            updateAction={() => console.warn('update')}
            removeAction={() => console.warn('remove')}
        />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
