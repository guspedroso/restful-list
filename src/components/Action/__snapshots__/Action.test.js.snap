// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action types snapshots work 1`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Disabled snapshots work 1`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={true}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`Disabled snapshots work 2`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`Hide snapshots work 1`] = `
<div
  className=""
/>
`;

exports[`Hide snapshots work 2`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`Open snapshots work 1`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open snapshots work 2`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`Open with different combinations of functions snapshots work 1`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 2`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 3`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 4`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 5`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 6`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Open with different combinations of functions snapshots work 7`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="action-col col"
      >
        <button
          className="action-button pull-right btn btn-secondary"
          disabled={false}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Outer class snapshots work 1`] = `
<div
  className="test"
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Edit
  </button>
</div>
`;

exports[`Snapshot works 1`] = `
<div
  className=""
>
  <button
    className="action-button pull-right btn btn-primary"
    disabled={false}
    type="button"
  >
    Edit
  </button>
</div>
`;
